src/Panel.jsx

```diff
@@ -23,22 +35,29 @@ var Panel = React.createClass({
     };
   },
 
-  handleSelect: function (e) {
-    if (this.props.onSelect) {
-      this._isChanging = true;
-      this.props.onSelect(this.props.eventKey);
-      this._isChanging = false;
+  handleSelect: function(e){
+    if(this.props.onSelect) {
+      this.props.onSelect(e, this.props.eventKey)
+    } else {
+      this.handleClickEvent(e);
+      this.handleToggle();
     }
+  },
 
+  handleClickEvent: function(e){
     e.preventDefault();
+  },
 
-    this.setState({
-      expanded: !this.state.expanded
-    });
+  handleToggle: function(){
```

Comment:
I think part of the reason why I removed the `expanded` tracking from the mixin is that it started to get confusing when you should use the `internal expanded` vs. the `external expanded`.  With only one set of state tracking in the mixin then it's very clear.